rust/connlib/clients/android/src/lib.rs

```diff
@@ -502,3 +497,29 @@ pub unsafe extern "system" fn Java_dev_firezone_android_tunnel_ConnlibSession_re
     let session = &*(session_ptr as *const SessionWrapper);
     session.inner.reconnect();
 }
+
+/// # Safety
+/// session_ptr should have been obtained from `connect` function, and shouldn't be dropped with disconnect
+/// at any point before or during operation of this function.
+#[allow(non_snake_case)]
+#[no_mangle]
+pub unsafe extern "system" fn Java_dev_firezone_android_tunnel_ConnlibSession_setTun(
+    mut env: JNIEnv,
+    _: JClass,
+    session_ptr: jlong,
+    fd: RawFd,
+) {
+    let session = &*(session_ptr as *const SessionWrapper);
+
+    // Enter tokio RT context to construct `Tun`.
+    let _enter = session.runtime.enter();
```

Comment:
Oh, cool I didn't know about this, does this also force execution within the runtimes context? The docs aren't very clear on this.

Also I think a comment about how this works because we have a single thread would be nice(or maybe I'm getting this wrong?)

Reply:
No, I don't think it could force a thread into the runtime temporarily.
As far as I know it just sets up thread-local state so you can do things like `tokio::spawn` without a handle to the runtime.

I'm not sure what it does in this context. `mpsc` and `AsyncFd` don't need to be in a context (Right?) and I don't see any new `spawn` calls.

Reply:
> `AsyncFd` don't need to be in a context

`AsyncFd` does need a context, it panics otherwise!